UI/src/view/LinearAssetLayer.js

```diff
@@ -334,20 +429,25 @@ window.LinearAssetLayer = function(params) {
     decorateSelection();
   };
 
+  var refreshReadOnlyLayer = function () {
+    if(massLimitation)
+      readOnlyLayer.refreshView();
+  };
+
   this.layerStarted = function(eventListener) {
     bindEvents(eventListener);
   };
 
-  this.refreshView = function(event) {
+  this.refreshView = function() {
     vectorLayer.setVisible(true);
     adjustStylesByZoomLevel(map.getView().getZoom());
     if (isComplementaryChecked) {
       collection.fetchAssetsWithComplementary(map.getView().calculateExtent(map.getSize())).then(function() {
-        eventbus.trigger('layer:linearAsset:' + event);
+        eventbus.trigger('layer:linearAsset');
```

Comment:
Why is refreshReadOnlyLayer called through the eventbus and not directly?

Reply:
To call it when we already had the others linearAssets